hack/update-bazel.sh

```diff
@@ -26,13 +26,14 @@ kube::util::ensure-gnu-sed
 # TODO(spxtr): Remove this line once Bazel is the only way to build.
 rm -f "${KUBE_ROOT}/pkg/generated/openapi/zz_generated.openapi.go"
 
-# The git commit sha1s here should match the values in $KUBE_ROOT/WORKSPACE.
-kube::util::go_install_from_commit \
-    github.com/kubernetes/repo-infra/kazel \
-    97099dccc8807e9159dc28f374a8f0602cab07e1
-kube::util::go_install_from_commit \
-    github.com/bazelbuild/bazel-gazelle/cmd/gazelle \
-    0.10.1
+# Ensure that we find the binaries we build before anything else.
+export GOBIN="${KUBE_OUTPUT_BINPATH}"
```

Comment:
Do we need to export this? I don't like modifying a user's environment beyond the life of the script. There's a couple other cases that you're doing this too, so I'm not sure if it's strictly a requirement.

Reply:
`export` exists for the life of the shell, not longer.  Maybe I misunderstand.  I did not put it in a generic place that gets sourced, for this reason.

```
$ echo $FOO; (export FOO=foo; echo $FOO; (export FOO=bar; echo $FOO); echo $FOO); echo $FOO

foo
bar
foo

```